src/parser/chords_over_words_grammar.pegjs

```diff
@@ -0,0 +1,128 @@
+ChordSheet
+  = metaDataLines:MetaData? lines:ChordSheetContents? {
+      return {
+        type: "chordSheet",
+        lines: [
+          ...metaDataLines,
+          ...lines,
+        ]
+      };
+    }
+
+ChordSheetContents
+  = NewLine? items:ChordSheetItem* {
+    return items;
+  }
+
+ChordSheetItem
+  = item:(InlineMetaData / ChordLyricsLines / LyricsLine) NewLine {
+    return item;
+  }
+
+ChordLyricsLines
+  = chords:ChordsLine NewLine lyrics:Lyrics {
+      const chordLyricsPairs = chords.map((chord, i) => {
+         const nextChord = chords[i + 1];
+         const start = chord.column - 1;
+         const end = nextChord ? nextChord.column - 1 : lyrics.length;
+
+         return {
+           type: "chordLyricsPair",
+           chord,
+           lyrics: lyrics.substring(start, end)
+         };
+      });
+
+      const firstChord = chords[0];
+
+      if (firstChord && chords[0].column > 0) {
+      	const firstChordPosition = firstChord.column;
+
+        if (firstChordPosition > 0) {
+          chordLyricsPairs.unshift({
+            type: "chordLyricsPair",
+            chord: null,
+            lyrics: lyrics.substring(0, firstChordPosition - 1),
+          });
+        }
+      }
+
+      return { type: "line", items: chordLyricsPairs };
+    }
```

Comment:
This looks good and is a similar approach to what I took in the [onsong](https://github.com/PraiseCharts/ChordSheetJS/pull/272/files#diff-53de6b9948a1bd529a53991dc298359b7b4ab4ca3c1f74361bb174a17947967cR22) parser. You could maybe eliminate the duplicate logic by unshifting an empty chord onto the `chords` array if the first chord does not start at column 0 prior to mapping chords.